### PR TITLE
Fix build profiles

### PR DESCRIPTION
**Pull request by ggwpez**
Closes https://github.com/paritytech/polkadot-sdk/issues/1155

Changes:
- Move the build profiles to the workspace. Otherwise they are ignored.
- Use keys `edition`, `authors` and `repository` from the workspace, where applicable.

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -448,15 +455,78 @@ members = [
...
 
-[workspace.package]
-authors = ["Parity Technologies <admin@parity.io>"]
-edition = "2021"
-repository = "https://github.com/paritytech/polkadot-sdk.git"
-license = "GPL-3.0-only"
-version = "1.0.0"
+[profile.release]
+# Polkadot runtime requires unwinding.
+panic = "unwind"
```
Comment (ggwpez) [inline Cargo.toml:460]: I guess this is generally fine?

Comment (bkchr) [reply Cargo.toml:460]: Should probably be the default anyway and was only required for the native runtime

File: Cargo.toml [hunk 2]
```diff
@@ -1,3 +1,10 @@
+[workspace.package]
+authors = ["Parity Technologies <admin@parity.io>"]
+edition = "2021"
+repository = "https://github.com/paritytech/polkadot-sdk.git"
+license = "GPL-3.0-only"
```
Comment (bkchr) [inline Cargo.toml:5]: The ones that are GPL3 with exception are not using the workspace license?

File: Cargo.toml [hunk 3]
```diff
@@ -1,3 +1,10 @@
...
+version = "1.0.0"
```
Comment (bkchr) [inline Cargo.toml:6]: Also not sure about this. We have a lot of crates in Substrate that don't use this version as they are already on a much higher version.

File: Cargo.toml (hunk 2 above)
Comment (ggwpez) [reply Cargo.toml:5]: Yes, its the same as before for those. But wen can also remove it and make it explicit for each crate :man_shrugging:

File: Cargo.toml (hunk 3 above)
Comment (ggwpez) [reply Cargo.toml:6]: I only see crates in Polkadot use the `version.workspace = true` syntax. `rococo-runtime` for example.  
Dont know what to do with these. Maybe setting them manually to `0.0.1`? @Morganamilo would then notice if we already released higher versions of those i guess.

Comment (bkchr) [reply Cargo.toml:6]: Yeah I know that we used this in Polkadot, but now we can not really do this anymore.

Comment (ggwpez) [reply Cargo.toml]: Okay I set it to `1.0.0` manually and removed the `version.workspace`. Now also added `edition`, `authors` and `repository`.

**Review by acatangiu — APPROVED** [review]

**Review by bkchr — APPROVED** [review]

**Pull request merged**